lua/neogit/client.lua

```diff
@@ -81,4 +81,33 @@ function M.editor(target, client)
   end
 end
 
+---@param cmd any
+---@param opts table
+function M.wrap(cmd, opts)
```

Comment:
I suppose this is a shorthand for running a command with the appropriate envs and and status notifications.

Perhaps `wrap_envs` signals the meaning slightly better, as `async.wrap` is already used quite a lot in the same code, so the `wrap` symbol is already occupied for a different purpose in the cognitive space.

Anyhow, it is good as is; just leaving for future selves. I'll go through and sprinkle some comments and type annotations time to time.

Reply:
Sure, makes sense to differentiate async wrap from env wrap